external/fv3fit/fv3fit/keras/math.py

```diff
@@ -0,0 +1,42 @@
+from typing import Callable
+import tensorflow as tf
+
+
+def piecewise(x: tf.Tensor, y: tf.Tensor, xg: tf.Tensor) -> tf.Tensor:
+    """1D 0th order interpolation
+
+    Args:
+        x: original ordinate, 1D, must be sorted
+        y:  original value, 1D
+        xg: points to interpolate at
+
+    Return:
+        yg: interpolated values. same shape as xg. Uses constant extrapolation.
+            Within the min/max of x it has the following form::
+
+                f(xg) = y[i] if x[i] <= xg < x[i+1]
+    """
+    xg_flat = tf.reshape(xg, [-1])
+    index = tf.searchsorted(x, xg_flat, side="right") - 1
```

Comment:
Will this raise an error if `x` is not actually sorted, or just silently return the wrong output?

Reply:
It may silently fail, but I decided to let tensorflow make it's own complaints if it wants to. I did put an assertion in the `groupby` to ensure we don't do that in practice.